### DISCUSSION TITLE
blog and apostrophe 3

### DISCUSSION THREAD

**Discussion by atoy40** [apostrophecms/apostrophe#3693] [General]
Hello,

I'm planning to migrate an apostrophe2 site to apostrophe3 but I'm using the apostrophe-blog module. It seems it's not yet exists for apostrophe3. It is something planned ? or is there a different/better way to handle blog posts in apostrophe3 ?

Another question, I'm also using the events module, this one seems to be ported to apo3, but is it compatible with the apo2 version in term of data in the db (and after a data migration for sure)

Thks
Anthony.

**Comment by boutell** [comment]
Hello Anthony,

There is no official port of the blog module yet. However if you take a look you will find the blog module source code is actually quite small. Really the only thing it adds over generic pieces is a default sort in reverse chronological order on a custom `publishedAt` field, and logic to respect that field when deciding what to show. If you don't need `publishedAt` you might be happy with just configuring the `sort` option of a piece type called `article` and creating a corresponding `article-page` module extending `@apostrophecms/piece-page-type`.

Or you could port `apostrophe-blog` as a community contribution. Community member `bulldoguk` already ported the similar and significantly more complicated `apostrophe-events` module as `@apostrophecms/event`. You could look at that module for a model to follow in porting the blog module.

**Comment by atoy40** [comment]
Thanks @boutell ,

I think I'll try to port the blog module to the new API, seems to be a good exercice to introduce a3 changes :)
Just a question to be sure I'm right. If you have an a3 piece with the same name and fields as it was in the a2 site, and you migrate the database, it just works transparently ?

 Anthony.

**Comment by boutell** [comment]
That should work, yes, if you use the official A3 content migration tool.

On Tue, Mar 29, 2022 at 3:36 PM Anthony Hinsinger ***@***.***>
wrote:

> Thanks @boutell <https://github.com/boutell> ,
>
> I think I'll try to port the blog module to the new API, seems to be a
> good exercice to introduce a3 changes :)
> Just a question to be sure I'm right. If you have an a3 piece with the
> same name and fields as it was in the a2 site, and you migrate the
> database, it just works transparently ?
>
> Anthony.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/apostrophecms/apostrophe/discussions/3693#discussioncomment-2464441>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAH27IOWSE46FO6UVMPFELVCNLSHANCNFSM5RTZUL6A>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***
> com>
>


-- 

THOMAS BOUTELL | CHIEF TECHNOLOGY OFFICER
APOSTROPHECMS | apostrophecms.com | he/him/his
